kernel/src/snapshot.rs

```diff
@@ -152,6 +152,64 @@ impl Snapshot {
     pub fn into_scan_builder(self) -> ScanBuilder {
         ScanBuilder::new(self)
     }
+
+    /// Update the `Snapshot` to the target version.
+    ///
+    /// # Parameters
+    /// - `target_version`: desired version of the `Snapshot` after update, defaults to latest.
+    /// - `engine`: Implementation of [`Engine`] apis.
+    ///
+    /// # Returns
+    /// - boolean flag indicating if the `Snapshot` was updated.
+    pub fn update(
+        &mut self,
+        target_version: impl Into<Option<Version>>,
+        engine: &dyn Engine,
+    ) -> DeltaResult<bool> {
+        let fs_client = engine.get_file_system_client();
+        let log_root = self.table_root.join("_delta_log/")?;
+        let log_segment = match LogSegment::for_table_changes(
+            fs_client.as_ref(),
+            log_root,
+            self.version() + 1,
+            target_version,
+        ) {
+            Ok(segment) => segment,
+            Err(Error::FileNotFound(_)) => return Ok(false),
+            Err(e) => return Err(e),
+        };
+
+        let (metadata, protocol) = log_segment.read_metadata_opt(engine)?;
+        if let Some(p) = &protocol {
+            p.ensure_read_supported()?;
```

Comment:
Should the column mapping mode also be checked in the case only Protocol is updated? the `column_mapping_mode` function takes protocol as a parameter, so I imagine whatever the output column_mapping_mode is should be re-validated.

Reply:
In fact this is another usecase for my [TableConfiguration PR](https://github.com/delta-io/delta-kernel-rs/pull/644) where we do all the validations in one place without having to think about these cases.